sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -308,7 +308,8 @@ static int _fill_ipv6_hdr(kernel_pid_t iface, ng_pktsnip_t *ipv6,
 
         /* We deal with multiple interfaces here (multicast) => possible
          * different source addresses => duplication of payload needed */
-        while (ptr != payload) {
+        while (ptr != payload->next) {
```

Comment:
For sending it is the upper layer (the inner header). I check for `payload->next` so the payload header gets duplicated too. Since the checksum field is in there and the checksum might be different (due to different source addresses in IPv6) this is necessary.